Automattic/jetpack-boost-critical-css-gen
# Port to TypeScript

This PR ports the library to TypeScript. When building, two JavaScript versions are built; `lib/` contains a simple `tsc`-compiled version of the library along with the TypeScript headers to allow consumers to use its types. `dist/` contains a web-ready bundled version of this library.

Along the way, it also: 
- Adds linting, and lints all of the files. 
- Adds lots of type information to existing functionality.

This PR doesn't:
- Refactor the system. This PR is all about paving the way for nicer and safer refactors in future PRs.

Testing:
- Make sure the tests all run properly.
- Make sure you can use the command line utility inbuilt; `npm run gen -- https://jetpack.com`
- Make sure it still works with Jetpack Boost (branch already setup to test here: https://github.com/Automattic/jetpack/tree/update/critical-css-gen-version
- Make sure it still works with the `update/css-generator` branch in `boost-cloud`.
